### ISSUE TITLE
Invalid responses from other nodes' /partyinfo endpoints are fatal

### ISSUE THREAD

**Issue by stevenroose** [jpmorganchase/constellation#39]
We're trying to get Constellation running in RHEL. Sadly Red Hat is not one of the major platforms where Constellation has precompiled binaries for :)

We compiled it on a machine with internet access (which was not trivial given a weird bunch of dependencies), but when trying to run it on another machine, we get the following error:

```
2017 Jun-21 13:51:38115940 [WARN] Failed to run an AtExit hook: : Data.Binary.Get.runGet at position 8: not enough bytes
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:342:5 in binary-0.8.3.0:Data.Binary.Get
constellation-node: Data.Binary.Get.runGet at position 8: not enough bytes
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:342:5 in binary-0.8.3.0:Data.Binary.Get
```

**Comment by patrickmn** [comment]
So it works fine on one RHEL 7.2 box, but not the other? What's the difference between the two boxes?

You might want to try building a statically linked (using musl) binary using `build-linux-static.dockerfile`, though as far as I know there are no known problems compiling and running normally on RHEL 6 and 7.

**Comment by patrickmn** [comment]
Also definitely agree that it should be easier to install (and keep up to date) binaries for both Constellation and Quorum. Issues to track: https://github.com/jpmorganchase/constellation/issues/40 and https://github.com/jpmorganchase/quorum/issues/133

**Comment by stevenroose** [comment]
Here is a more detailed log:
```
$ ./constellation-node Ethereum/julionet/constellation.conf -vvv
2017 Jun-21 15:54:17451181 [INFO] Log level is LevelDebug
2017 Jun-21 15:54:17451413 [DEBUG] Configuration: Config {cfgUrl = "http://[censored]:8006/", cfgPort = 8006, cfgSocket = Just "/home/p66188/Ethereum/julionet/constellation.ipc", cfgOtherNodes = ["[censored]"], cfgPublicKeys = ["/home/p66188/Ethereum/julionet/constellation-pub"], cfgPrivateKeys = ["/home/p66188/Ethereum/julionet/constellation-prv"], cfgAlwaysSendTo = [], cfgPasswords = Nothing, cfgStorage = "/home/p66188/Ethereum/julionet/constellation", cfgIpWhitelist = [], cfgJustShowVersion = False, cfgJustGenerateKeys = [], cfgVerbosity = 5}
2017 Jun-21 15:54:17451731 [INFO] Utilizing 2 core(s)
2017 Jun-21 15:54:17451904 [INFO] Constructing Enclave using keypairs [("/home/p66188/Ethereum/julionet/constellation-pub","/home/p66188/Ethereum/julionet/constellation-prv")]
Unlocking /home/p66188/Ethereum/julionet/constellation-prv
Unlocked /home/p66188/Ethereum/julionet/constellation-prv
2017 Jun-21 15:54:17452671 [INFO] Initializing storage /home/p66188/Ethereum/julionet/constellation
2017 Jun-21 15:54:17595363 [INFO] Public API listening on 0.0.0.0 port 8006 with whitelist: ["Disabled"]
2017 Jun-21 15:54:17595613 [INFO] Internal API listening on /home/p66188/Ethereum/julionet/constellation.ipc
2017 Jun-21 15:54:17595223 [INFO] Node started
2017 Jun-21 15:54:17596078 [INFO] Starting synchronization with http://180.46.5.212:8006/
2017 Jun-21 15:54:17599954 [INFO] Finished synchronization with http://180.46.5.212:8006/
2017 Jun-21 15:54:17599998 [INFO] Starting synchronization with http://180.46.5.213:8006/
2017 Jun-21 15:54:17603926 [INFO] Finished synchronization with http://180.46.5.213:8006/
2017 Jun-21 15:54:17603979 [INFO] Starting synchronization with http://180.46.5.216:8006/
2017 Jun-21 15:54:17606367 [INFO] Finished synchronization with http://180.46.5.216:8006/
2017 Jun-21 15:54:17606416 [INFO] Starting synchronization with http://180.46.5.209:8006/
2017 Jun-21 15:54:17608966 [INFO] Finished synchronization with http://180.46.5.209:8006/
2017 Jun-21 15:54:17609015 [INFO] Starting synchronization with http://180.46.5.210:8006/
2017 Jun-21 15:54:17611982 [INFO] Finished synchronization with http://180.46.5.210:8006/
2017 Jun-21 15:54:17612022 [INFO] Starting synchronization with http://180.46.5.217:8006/
2017 Jun-21 15:54:17614305 [INFO] Finished synchronization with http://180.46.5.217:8006/
2017 Jun-21 15:54:17614348 [INFO] Starting synchronization with http://180.46.5.211:8006/
2017 Jun-21 15:54:17616984 [INFO] Finished synchronization with http://180.46.5.211:8006/
2017 Jun-21 15:54:17617028 [INFO] Starting synchronization with http://180.46.5.214:8006/
2017 Jun-21 15:54:17619860 [INFO] Finished synchronization with http://180.46.5.214:8006/
2017 Jun-21 15:54:18620392 [INFO] Shutting down... (Interrupting this will cause the next startup to take longer)
2017 Jun-21 15:54:18620605 [WARN] Failed to run an AtExit hook: : Data.Binary.Get.runGet at position 8: not enough bytes
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:342:5 in binary-0.8.3.0:Data.Binary.Get
constellation-node: Data.Binary.Get.runGet at position 8: not enough bytes
CallStack (from HasCallStack):
  error, called at libraries/binary/src/Data/Binary/Get.hs:342:5 in binary-0.8.3.0:Data.Binary.Get
```

**Comment by patrickmn** [comment]
Are you shutting it down (or is it being sent a kill signal) prior to receiving that Binary error message?

Are you running the same version of Constellation on all the nodes?

Please provide a reproducible example if possible (binaries+configs.)

**Comment by stevenroose** [comment]
I'm trying the static compilation now with stack. But to answer your questions: we're not shutting it down manually. We are running just one node, the others are unavailable, but that should not be a problem, should it? I guess it should just keep trying to find them.

We'll report later and try to get a reproducible example.

**Comment by patrickmn** [comment]
If for some reason one of those hosts is actually responding to POST requests to `<host>/partyinfo`, that could be the source of the problem. Your node would be trying to decode the response body from a non-Constellation response. No response/connection refused shouldn't cause a problem.

**Comment by stevenroose** [comment]
Yeah, that's what we just realized. They are proxied. All the machines within the bank firewall use a system-wide proxy. So the proxy probably doesn't find the machines but still behaves differently than an IP that does not listen to that port.

This is a big problem. When setting up multiple machines, the first will always run without the other one already being there. So this is fundamentally incompatible with proxy usage.

**Comment by patrickmn** [comment]
Ok, I'm glad that it's not some mysterious linking issue.

It's simple to "fix" this; we can just ignore decode errors here. Will push a PR in a bit.

**Comment by patrickmn** [comment]
Can you please tell me if recompiling from https://github.com/jpmorganchase/constellation/pull/41 fixes the issue for you?

**Comment by stevenroose** [comment]
We will do tomorrow :) It's time to go home here in Spain. Thanks a lot for the swift responses!!

**Comment by patrickmn** [comment]
Sure thing :)

**Comment by stevenroose** [comment]
Fixed by #41! Can be closed when it lands.
